### DISCUSSION TITLE
Z2M high CPU load in Docker

### DISCUSSION THREAD

**Discussion by Olivierr92500** [Koenkk/zigbee2mqtt#10975] [Devices]
Hello,

I am using Z2M together with OpenHab 3 on a Raspberry Pi 3B+ for my home automation setup.
I have installed Z2M  release 1.22.1 commit: 89996835 using docker on a Raspbian OS 10 (Buster)
I am using a  CC2652 BLE based coordinator (zStack3x0) revision 20210708 with interface version 0.6.46

My docker node is using 100% of 1 of the 4 cores (25% of CPU) with around 25 Zigbee devices.
I do not think my previous controller (CC2531) consumption was so high but I did not measure.

What CPU consumption do you have? I have no reference but 1 full core seams too much.

Kind regards

Olivier

**Comment by boriken72** [comment]
i am seeing the same thing.  I am on build 1.23.0 commit: 6ea61ba using docker on debian 11. I am using slaeh's CC2652RB stick (USB-A) as coordinator . I have currently 42 devices.  This seems to happen after running for couple days.   Consumes 100% of 1 core and i have to restart to recover.  The only way i'm aware is because i am integrated with home assistant and it will continuously notify me of device state (open/closed) which typically only happens when I restart zigbee2mqtt container.

**Comment by mikeq3** [comment]
Has this issue with high cpu been resolved? I'm currently having the same issue with the latest and previous docker image. node consuming 100% of one core.

**Comment by wgentine** [comment]
Same issue here... so its not an isolated problem. 1.29.1 version

**Comment by shackrat** [comment]
This is still a thing in 1.29.2 as well.  I have three instances on Z2M running on a Raspberry Pi 4 (4GB).  It serves as a remote radio server hosting ZWaveJS in addition to the 3 instances of Z2M.  All of the Zigbee dongles are identical Sonoff ZBDongle-P all running identical firmware builds.

Z2M #1 is on channel 11 and has 101 devices.
Z2M #2 is on channel 12 and has 104 devices.
Z2M #3 is on channel 14 and has 3 problematic Lightify bulbs.

Only Instance #2 is affected by this issue.  Instances 1 & 3 are fine.  Instance #2 is affected roughly every 24 hours. 

Here's a chart from the Home Assistant sensors that monitor the rPi..   The runaway started around 2am with steadily increasing memory usage and eventually CPU usage, likely due to swapping.  I restarted the container around 8:25am today.

<img width="984" alt="image" src="https://user-images.githubusercontent.com/18385917/215933196-4de8e109-bdfa-4b96-aa39-62feef7a914d.png">

One interesting thing about the crashing instance, all device bindings were to endpoint 242 (Zigbee Green Power), #1 & #3 bindings were to endpoint 1 as expected.  Newly paired devices have been binding to endpoint 1, so I just completed re-pairing all devices to ensure bindings are correct.  Going to monitor for a couple days and see if that helps.
